airflow/www/views.py

```diff
@@ -156,12 +158,19 @@ class AirflowBaseView(BaseView):
     }
 
     def render_template(self, *args, **kwargs):
-        return super().render_template(
+        templ = super().render_template(
             *args,
             # Cache this at most once per request, not for the lifetime of the view instance
             scheduler_job=lazy_object_proxy.Proxy(jobs.SchedulerJob.most_recent_job),
             **kwargs
         )
+        resp = make_response(templ)
+        for k, v in kwargs.get('cookies', {}).items():
+            if v:
+                resp.set_cookie(k, v, expires=2147483647)
```

Comment:
What is this doing? What is the magic number here?

Can we instead make use of the existing Flask session and store something in there?

Reply:
I'm trying to save the users filters as long as possible (for example, if I work on the dags of team A, this is all I want to see, and I don't want to change the filters every day), so the magic number is max int.

I can use flask session but flask session doesn't support setting expiration per cookie, only a global setting (`permanent_session_lifetime`) and I don't think it's best practice.

Reply:
Changed to flask session.
Just and FYI - this change now makes the login session permanent as well (I know it's not an API change, but it's a security backwards-incompatible changes, so maybe I should write it in `UPDATING.md`?).